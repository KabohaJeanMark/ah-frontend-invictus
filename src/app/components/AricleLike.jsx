// import React from 'react';
// import PropTypes from 'prop-types';

// const ArticleLike = ({ handleLike, handleDislike }) => {
//   return (

//         <>
//         <div className="like-article">
//           <div className="row">
//             <div className="col-md-2" />
//             <div className="col-md-8">
//               <div className="card">
//                 <div className="like-dislike">
//                   <i
//                     className="far fa-thumbs-up"
//                     onClick={handleLike}
//                     role="presentation"
//                     style={{ color: 'blue' }
//                     }
//                   />
//                   <i
//                     className="far fa-thumbs-down"
//                     onClick={handleDislike}
//                     role="presentation"
//                     style={{ color: 'red' }}
//                   />
//                 </div>
//               </div>

//             </div>
//             <div className="col-md-2" />
//           </div>
//         </div>

//     </>
//   );
// };

// ArticleLike.propTypes = {
//   handleLike: PropTypes.func,
//   handleDislike: PropTypes.func
// };

// export default ArticleLike;
