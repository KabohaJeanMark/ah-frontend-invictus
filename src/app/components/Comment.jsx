import React from 'react';
import PropTypes from 'prop-types';
import '../../assets/styles/login.scss';


const Comment = ({ handleSubmit, handleChange }) => {
  return (
    <div>
      <div className="row">
        <div className="col-md-5" />
        <form onSubmit={handleSubmit}>
          <label>Comment</label>
          <textarea id="CommentBody" onChange={handleChange}
            className="form-control input-field" name="commentBody"
            placeholder="Enter your Comments" />
          <input type="submit" className="btn btn-primary login-btn" />
        </form>
      </div>
    </div>
  );
};

Comment.propTypes = {
  handleSubmit: PropTypes.func,
  handleChange: PropTypes.func
};

export default Comment;
