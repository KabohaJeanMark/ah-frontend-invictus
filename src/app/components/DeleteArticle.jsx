import React from 'react';

export default function DeleteArticleComponent (props) {
  return (
    <div >
      <button className="btn btn-outline-danger"
        onClick={props.handleDelete}>Delete</button>
    </div>
  );
}
