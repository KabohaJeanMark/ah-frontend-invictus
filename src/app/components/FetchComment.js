import React from 'react';
import PropTypes from 'prop-types';
import '../../assets/styles/login.scss';
import '../../assets/styles/fetchcomment.scss';

const FetchComment = ({ comments }) => {
  return (
    <>
      <div className="row content">
        <div className="col-md-12">
          {
            comments.map(comment => {
              return (
                <div className="input-field" >
                  {comment.body}
                  <br />
                  {comment.created_at}
                  <br />
                  {comment.author}
                </div>
              );
            })
          }

        </div>
      </div>
    </>
  );
};

FetchComment.propTypes = {
  comment: PropTypes.string,
  comments: PropTypes.array
};

FetchComment.defaultProps = {
  comments: []
};

export default FetchComment;
