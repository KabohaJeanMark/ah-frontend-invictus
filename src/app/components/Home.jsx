import React, { Component } from 'react';
import NavbarComponent from '../components/Navbar';
import Articles from '../containers/articles/ArticlesContainer';

class Home extends Component {
  render () {
    return (
      <React.Fragment>
        <NavbarComponent />
        <Articles />
      </React.Fragment>
    );
  }
}

export default Home;
