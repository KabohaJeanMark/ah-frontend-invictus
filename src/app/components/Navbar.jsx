
import React, { Component } from 'react';
import '../../assets/styles/navbar.scss';
import { Link } from "react-router-dom";
import { connect } from "react-redux";
export class NavbarComponent extends Component {
  render () {
    const { isLoggedIn } = this.props;
    return (
      <React.Fragment>
        <nav className="navbar navbar-expand-lg navbar-dark">
          <button className="navbar-toggler nav-button" type="button"
            data-toggle="collapse" data-target="#navbarTogglerah"
            aria-controls="navbarTogglerDemo01" aria-expanded="false"
            aria-label="Toggle navigation">
            <span className="navbar-toggler-icon" /></button>
          <div className="collapse navbar-collapse" id="navbarTogglerah">
            <ul className="navbar-nav mr-auto mt-2 mt-lg-0">
              <li className="nav-item active">
                <Link className="nav-link" to="/">Home</Link></li>
            </ul>
            {isLoggedIn ? (
              <ul className="navbar-nav ml-2 ">
                <li className="nav-item">
                  <Link className="nav-link" to="/articles">Articles</Link></li>
                <li className="nav-item" />
                <li className="nav-item">
                  <Link className="nav-link" to="/profile">Profile</Link></li>
                <li className="nav-item">
                  <Link className="nav-link" to="/login">Logout</Link></li>
              </ul>) : (
              <ul className="navbar-nav ml-2 ">
                <li className="nav-item">
                  <Link className="nav-link" to="/login">Login </Link></li>
                <li className="nav-item">
                  <Link className="nav-link nav-link-active" to="signup">Signup </Link>
                </li>
              </ul>)}
          </div>
        </nav >
      </React.Fragment >
    );
  }
}

export const mapStateToProps = (state) => {
  const { username, isLoggedIn } = state.auth;
  return { username, isLoggedIn };
};

export default connect(mapStateToProps)(NavbarComponent);

