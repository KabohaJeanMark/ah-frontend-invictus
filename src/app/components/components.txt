All app components will go into this folder.
Delete this after adding atleast one component.
