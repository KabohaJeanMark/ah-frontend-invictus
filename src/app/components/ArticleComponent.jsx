import React from 'react';
import { WithContext as ReactTags } from 'react-tag-input';
import '../../assets/styles/article.scss';
import Navbar from '../components/Navbar';

const KeyCodes = {
  comma: 188,
  enter: 13
};

const delimiters = [KeyCodes.comma, KeyCodes.enter];

const ArticlePresentationComponent = ({ handleSubmit, handleChange, handleAddTag, handleDeleteTag, tags }) => {
  return (
    <>
    <Navbar />
      <div className="create-article">
        <div className="row">
          <div className="col-md-2" />
          <div className="col-md-8">
            <div className="card">
              <form onSubmit={handleSubmit}>
                <div className="form-group">
                  <label className="sr-only">Title</label>
                  <textarea
                    className="title-input"
                    type="text"
                    name="title"
                    id="title-input"
                    placeholder="Title"
                    required
                    onChange={(e) => handleChange(e)}/>
                </div>
                <div className="form-group">
                  <textarea
                    className="description-input"
                    type="text"
                    name="description"
                    id="description-input"
                    placeholder="Description"
                    required
                    onChange={(e) => handleChange(e)}
                  />
                </div>
                <div className="form-group">
                  <textarea
                    className="body-input"
                    type="text"
                    name="body"
                    id="body-input"
                    placeholder="Body"
                    required
                    onChange={e => handleChange(e)}
                  />
                </div>
                <div className="form-group">
                  <label htmlFor="tags">Tags</label>
                  <ReactTags
                    classNames="tags"
                    maxLength = "42"
                    inline
                    tags={tags}
                    handleAddition={handleAddTag}
                    handleDelete={handleDeleteTag}
                    delimiters={delimiters}
                  />
                </div>
                <button type="submit" className="btn postarticle-btn btn-outline-primary">Post Article</button>
              </form>
            </div>
          </div>
          <div className="col-md-2" />
        </div>
      </div>
    </>
  );
};

export default ArticlePresentationComponent;
