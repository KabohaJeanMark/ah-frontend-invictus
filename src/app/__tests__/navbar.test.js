import React from 'react';
import { shallow } from "enzyme";
import {NavbarComponent, mapStateToProps } from "../components/Navbar";
import toJson from 'enzyme-to-json';

describe('<NavbarComponent/>', () => {
  const props = {
  
        isLoggedIn: false, username:""
  };
  let wrapper;
  it('renders navbar component', () => {
    wrapper = shallow(<NavbarComponent {...props} />);
  });
  it('matches snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it ('should map state to props', ()=>{
    const mockState = {auth:{
      username:'edna',
      isLoggedIn: true
    }};
    const state = mapStateToProps(mockState)
    expect(state.username).toEqual('edna')
    expect(state.isLoggedIn).toEqual(true)
  });
});

