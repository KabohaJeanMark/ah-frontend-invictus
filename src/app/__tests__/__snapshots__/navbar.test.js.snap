// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavbarComponent/> matches snapshot 1`] = `
<Fragment>
  <nav
    className="navbar navbar-expand-lg navbar-dark"
  >
    <button
      aria-controls="navbarTogglerDemo01"
      aria-expanded="false"
      aria-label="Toggle navigation"
      className="navbar-toggler nav-button"
      data-target="#navbarTogglerah"
      data-toggle="collapse"
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse"
      id="navbarTogglerah"
    >
      <ul
        className="navbar-nav mr-auto mt-2 mt-lg-0"
      >
        <li
          className="nav-item active"
        >
          <Link
            className="nav-link"
            to="/"
          >
            Home
          </Link>
        </li>
      </ul>
      <ul
        className="navbar-nav ml-2 "
      >
        <li
          className="nav-item"
        >
          <Link
            className="nav-link"
            to="/login"
          >
            Login 
          </Link>
        </li>
        <li
          className="nav-item"
        >
          <Link
            className="nav-link nav-link-active"
            to="signup"
          >
            Signup 
          </Link>
        </li>
      </ul>
    </div>
  </nav>
</Fragment>
`;
